src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -233,6 +233,12 @@ public void BinaryOperatorOverloadResolution(BinaryOperatorKind kind, BoundExpre
 
                 foreach (NamedTypeSymbol @interface in interfaces)
                 {
+                    if (!@interface.IsInterface)
+                    {
+                        // this code could be reachable in error situations
+                        continue;
```

Comment:
No, we are not. But we are checking for this case in other similar situations, for example in TypeSymbolExtensions.ImplementsInterface. I am sure we can hit them here, but I don't think it is worth coming up with a test scenario that would have an observable effect in behavior.